### PR TITLE
Localization

### PR DESCRIPTION
**Pull request by gomezc**

### CODE REVIEW

File: launch/sicknav350.launch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<launch>
+  <node name="static_tf0" pkg="tf" type="static_transform_publisher"
+        args="0 0 0 0 0 0 /world /map 100" />
+
+  <node name="sicknav350" pkg="sicktoolbox_wrapper" type="sicknav350" output="screen">
+    <rosparam>
+      ipaddress: 192.168.1.10 <!--Sick sensor ip address -->
```
Comment (shaun-edwards) [inline launch/sicknav350.launch:7]: This should be a launch parameter, not hard-coded.

File: ros/sicknav350/sicknav350.cpp [hunk 2]
```diff
@@ -65,228 +66,312 @@ void publish_scan(ros::Publisher *pub, double *range_values,
...
+    }
+  }
+  double get_mean() {
+    return sum/deq.size();
+  }
 };
-       
-void PublishReflectorTransform(double x,double y,double th,tf::TransformBroadcaster odom_broadcaster,std::string frame_id,std::string child_frame_id)
+
+void PublishReflectorTransform(std::vector<double> x,std::vector<double> y,double th,std::vector<tf::TransformBroadcaster> odom_broadcaster,std::string frame_id,std::string child_frame_id)
```
Comment (shaun-edwards) [inline ros/sicknav350/sicknav350.cpp:129]: vectors should be passed by reference..

File: CMakeLists.txt [hunk 3]
```diff
@@ -47,6 +47,7 @@ target_link_libraries(sicklms ${catkin_LIBRARIES})
 
 add_executable(sicknav350 ros/sicknav350/sicknav350.cpp)
 target_link_libraries(sicknav350 ${catkin_LIBRARIES})
+add_dependencies(sicknav350 ${PROJECT_NAME}_generate_messages_cpp)
```
Comment (shaun-edwards) [inline CMakeLists.txt:50]: This message doesn't seem useful.  Was it used before and you just fixed a `cmake` issue?

**Comment by shaun-edwards** [comment]
Without running the node, it feels like it might pollute the logs...lots of `ROS_INFO` calls in for loops.  If this is the case, we should change these to `ROS_DEBUG`

**Comment by shaun-edwards** [comment]
I'm not really sure how to review this.  The other packages only produce laser scans, so their ROS node wrappers are more straightforward.  It feels like a `nav350` node should simply produce a transform from world/map to it's own frame, although this is not what navigation expects.

Let's address the few issues I found and move on.  If/when we do a PR back to the main repo, it may not go smoothly.

**Comment by gavanderhoorn** [comment]
The `indigo-devel` branch does not have commit 2293937b from this PR, which makes `sicktoolbox_wrapper` fail to compile due to the message headers not being generated before compiling the `sicknav350` target.

The `rosinstall` file referenced in `README.md` checks out `indigo-devel`, so building the workspace fails right now.

**Comment by jrgnicho** [comment]
@gomezc is this a valid PR?

**Comment by gomezc** [comment]
@jrgnicho It was at the time, but I don't know that it is now. It looks like more commits have happened on the original repo (larics). If it never got merged into the main swri-robotics branch, then I would think no one ever used these changes. I can't recall what the intent was - I know I was troubleshooting the sensor and I don't know what the euler platform ended up using. If no one had need of these changes, then I can withdraw the PR.

**Comment by jrgnicho** [comment]
The euler demo ended up using the euler-integration branch, so it'd be safe to close this PR as no project that I'm aware of is using it.

**Pull request closed without merging**